bin/report

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+# bin/report <build-dir> <cache-dir> <env-dir>
+
+### Configure environment
+
+set -o errexit    # always exit on error
+set -o pipefail   # don't ignore exit codes when piping output
+
+BUILD_DIR=${1:-}
+
+# <separator> <array>
+joinArrayWithSeparator() {
+  local IFS="$1"
+  shift
+  echo "$*"
+}
+
+packages=()
+custom_packages=()
+custom_repositories=()
+
+while IFS= read -r line; do
+  if [[ $line ]]; then
+    if echo "${line}" | grep -q -e "^:repo:"; then
+      custom_repositories+=("${line//:repo:/}")
+    elif [[ $line == *deb ]]; then
+      custom_packages+=("${line}")
+    else
+      packages+=("${line}")
+    fi
+  fi
+done < <(grep -v -s -e '^#' < "${BUILD_DIR}/Aptfile")
+
+echo "packages: \"$(joinArrayWithSeparator "," "${packages[@]}")\""
```

Comment:
For each of these three `echo`s, you could do `"${variable[@]-"(none)"}"` for slightly nicer output.

Reply:
i believe the output is meant to be ingested as YAML so, while that would be nicer, i think we want to retain the `""` when the array is empty.

Reply:
i'm basing my YAML comment on [this](https://github.com/heroku/heroku-buildpack-nodejs/blob/main/bin/report#L51).

i added an `output_key_value` function which handles all the array sorting, joining, escaping + it also skips the output if the value is empty.